src/shared/utils/selection-manager/selection.manager.ts

```diff
@@ -64,8 +101,11 @@ export class UmbSelectionManager {
 	 * @memberof UmbSelectionManager
 	 */
 	public select(unique: string | null) {
+		if (this.getSelectable() === false) return;
+		if (this.isSelected(unique)) return;
 		const newSelection = this.getMultiple() ? [...this.getSelection(), unique] : [unique];
 		this.#selection.next(newSelection);
+		this._host.getHostElement().dispatchEvent(new UmbSelectionChangeEvent());
```

Comment:
Since this is a Controller it self, you can omit `._host` as the getHostElement() is recursive. (Its gets the parent controller that has an element)